core/classifiers/classifier_utils.py

```diff
@@ -60,15 +59,101 @@ def extract_svm_parameters(clf):
+
+    Raises:
+        Exception. If any of the strings in classifier data is not a unicode
+            string.
+    """
+    if isinstance(classifier_data, dict):
+        for k in classifier_data.keys():
             if isinstance(k, str):
                 raise Exception('Expected %s to be unicode but found str.' % k)
-            unicode_validator_for_classifier_data(var[k])
-    if isinstance(var, (list, set, tuple)):
-        for item in var:
+            unicode_validator_for_classifier_data(classifier_data[k])
+    if isinstance(classifier_data, list):
+        for item in classifier_data:
             unicode_validator_for_classifier_data(item)
-    if isinstance(var, str):
-        raise Exception('Expected \'%s\' to be unicode but found str.' % var)
+    if isinstance(classifier_data, str):
+        raise Exception(
+            'Expected \'%s\' to be unicode but found str.' % classifier_data)
+
+
+FLOAT_INDICATOR_KEY = 'float_values'
+
+
+# pylint: disable=too-many-branches
+def convert_float_numbers_to_string_in_classifier_data(classifier_data):
+    """Converts all floating point numbers in classifier data to string.
+
+    Args:
+        classifier_data: dict|list. The original classifier data which needs
+            conversion of floats to strings.
+
+    Returns:
+        Dict|List. Modified classifier data in which float values are converted
+            into strings and each dict/subdict is augmented with a special key
+            'float_values' which contains list of keys whose values have
+            undergone the transformation.
+    """
+    if isinstance(classifier_data, dict):
+        if FLOAT_INDICATOR_KEY in classifier_data:
+            raise Exception(
+                'Classifier data already contains a %s key' %
+                FLOAT_INDICATOR_KEY)
+        float_fields = []
+        for k in classifier_data:
+            if isinstance(classifier_data[k], (basestring, int)):
+                classifier_data[k] = classifier_data[k]
+            elif isinstance(classifier_data[k], dict):
+                classifier_data[k] = (
+                    convert_float_numbers_to_string_in_classifier_data(
+                        classifier_data[k]))
+            elif isinstance(classifier_data[k], list):
+                new_list, is_any_value_float = (
+                    convert_float_numbers_to_string_in_classifier_data(
+                        classifier_data[k]))
+                if is_any_value_float:
+                    float_fields.append(k)
+                classifier_data[k] = new_list
+            elif isinstance(classifier_data[k], float):
+                classifier_data[k] = str(classifier_data[k])
+                float_fields.append(k)
+            else:
+                raise Exception(
+                    'Expected all classifier data dict values to be dicts, '
+                    'lists, floats, integers or strings but received %s.' %(
```

Comment:
nit: add space after last '%'. If it causes a line-length-exceeded, change "integers" to "ints".